lib/httparty/connection_adapter.rb

```diff
@@ -69,10 +70,11 @@ def initialize(uri, options = {})
 
     def connection
       host = clean_host(uri.host)
+      default_port = uri.scheme == 'https' ? 443 : 80
```

Comment:
Definitely and the current source covers that.

While Ruby's URI ensures a value on the port attribute after its initialization, Addressable's URI doesn't set it when parsing a URL running on a default port (no mather it's https or http). That's why I've set this fallback.

```Ruby
require "minitest/autorun"
require "uri"
require "addressable/uri"

class TestRubyImplementations < Minitest::Unit::TestCase

  def test_uri_parse_on_default_ports
    url = 'https://foo/bar'
    uri = URI.parse url
    assert_equal uri.port, 443

    uri = Addressable::URI.parse url
    assert_equal uri.port, nil
  end

  def test_uri_parse_on_non_default_ports
    url = 'https://foo:8443/bar'
    uri = URI.parse url
    assert_equal uri.port, 8443

    uri = Addressable::URI.parse url
    assert_equal uri.port, 8443
  end

end
```
Hoping to make it clear, I'm DRYing up default port fallback.